qa/tests/src/org/jboss/jbossts/qa/Utils/JNDIManager.java

```diff
@@ -82,7 +81,12 @@ public static void main(String[] args) {
 
                 XADataSourceReflectionWrapper wrapper = new XADataSourceReflectionWrapper("oracle.jdbc.xa.client.OracleXADataSource");
 
-                wrapper.setProperty("databaseName", databaseName);
+                // The following `if` checks whether the service name should be used instead of the database name
+                if (!serviceName.equals("")) {
+                    wrapper.setProperty("serviceName", serviceName);
+                } else {
+                    wrapper.setProperty("databaseName", databaseName);
+                }
```

Comment:
I think it seems that we would read in `profileName + "_DatabaseName"` and either set it as databaseName or serviceName. Given that we don't try to read say `profileName + "_ServiceName"` we are basically using databaseName as one of two properties.

That being said I don't understand when serviceName would ever be "" but databaseName could have content - please can you clarify?

Reply:
Sorry, my mistake (refer to the other comment)